src/components/TodoList.tsx

```diff
@@ -1,19 +1,80 @@
-import { useState } from 'react';
-import { Container, Title, TextInput } from '@mantine/core';
+import { Button, Container, List, TextInput } from '@mantine/core';
+import type { ChangeEvent } from 'react';
+import { useEffect, useLayoutEffect, useRef, useState } from 'react';
 
 const TodoList = () => {
-  const [inputData, setInputData] = useState('');
-  const [items, setItems] = useState([]);
+  const isMounted = useRef(false);
+  const [task, setTask] = useState('');
+  const [todoList, setTodoList] = useState<Array<{ taskName: string; id: string }>>([]);
 
-  const addItem = () => {
-    // eslint-disable-next-line no-empty
-    if (!inputData) {
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setTask(event.target.value);
+  };
+
+  const handleDelete = (taskToDelete: unknown) => {
+    const deleted = todoList.filter((t) => t.id !== taskToDelete);
+    setTodoList(deleted);
+    localStorage.setItem('localTasks', JSON.stringify(deleted));
+  };
+
+  const addTask = () => {
+    setTodoList((prevTodoList) =>
+      prevTodoList.concat({
```

Comment:
Rather then concatenating the previous Todo list I would instead  do something like this 
```Javascript
setTodoList((prevTodoList) => [
  ...prevTodoList, 
  {
    taskName: task,
    id: new Date()
  } 
]);
```
Rather than modifying the object (`prevTodoList`) that is in memory we want to create a new object. This is to avoid any possible conflicts and allows your code to be forward compatible if for what ever reason `concat` is deprecated or replaced